Add support for ESM

mantoni opened the pull request:
Hi there! Thank you for this nice module. It worked nicely for me until I converted a project to ESM.

This PR introduces a `sourceType` option which is passed through to acorn and adds a skipped test for ESM. Unfortunately I'm not sure how to implement the AST transformation. Can you help out? 🙏

mantoni committed:
Add missing t.end() to tests

mantoni committed:
Add sourceType option for acorn

mantoni committed:
add test for esm

pirxpilot committed:
add support for droping imports

pirxpilot commented:
Have a look at my quick and dirty implementation. Maybe you could  see if that works for you. Add more tests if it does not. See if you can clean up the code if it does ;-)

mantoni commented:
Awesome! Thank you for the quick help ❤️

I realised that I also need to remove require statements if `sourceType = 'module'` because it's valid to use `require` in a module, and I might have bundled dependencies that are not ESM. I've added a test for that and also simplified the `drop` implementation.

Let me know if there is anything else I can do 🙏

pirxpilot commented:
Thanks. Should be released as 1.1.0 now.
I simplified a history a bit and kept an extra check for requires/imports before we start checking for a specific module. It's not necessary but since check for a specific require call is a bit elaborate and since that function is called on each AST node I figured we might as well keep it similar to what it was before this PR...
Thanks again.
